hibernate-core/src/main/java/org/hibernate/hql/internal/classic/QueryTranslatorImpl.java

```diff
@@ -1316,6 +1318,14 @@ protected boolean isSubselectLoadingEnabled() {
 		return hasSubselectLoadableCollections();
 	}
 
+	@Override
+	public List<String> getPrimaryFromClauseTables() {
+		return (List<String>) joins.values()
+				.stream()
+				.map( elem -> ((JoinSequence) elem).getFromPart().getFirstJoin().getJoinable().getTableName() ).
+						collect( Collectors.toList() );
```

Comment:
I don't think this is tested.

Is this `classic` thing of some use anyway?

Reply:
I think we should remove it. I tried to test it but it doesn't take an UPDATE clause over `createQuery, so there's no way I could write a meaningful test for the `classic` mode.

Reply:
If we expect this to never be called, I would prefer us throwing an `IllegalStateException` rather than adding code we can't test.

I would also lean towards removing the classic mode but I'm pretty sure it will be removed in 6 so let's keep it if it's not too much work.